docassemble/ALToolbox/test_al_income.py

```diff
@@ -0,0 +1,153 @@
+import unittest
+
+from decimal import Decimal
+from .al_income import (
+    ALIncome,
+    ALIncomeList,
+    ALAsset,
+    ALAssetList,
+    ALVehicle,
+    ALVehicleList,
+    ALSimpleValue,
+    ALSimpleValueList,
+    ALItemizedJob,
+    ALItemizedValueDict,
+    ALItemizedValue,
+    recent_years,
+)
+
+
+class test_correct_outputs(unittest.TestCase):
+    def test_simple_value(self):
+        value = ALSimpleValue(transaction_type="expense", value=5)
+        self.assertEqual(Decimal(-5), value.total())
+        value.transaction_type = "asset"
+        self.assertEqual(Decimal(5), value.total())
+
+        value_str = ALSimpleValue(value="5.3")
+        self.assertEqual(Decimal("5.3"), value_str.total())
+        value_neg_str = ALSimpleValue(transaction_type="expense", value="5.3")
+        self.assertEqual(Decimal("-5.3"), value_neg_str.total())
```

Comment:
Are we trying to test that a `Decimal` value is returned. If not, I think this may not be as clear as it could be because it uses `Decimal()`. I'm ok with whatever choice you make, but if we wanted to we could do something like:
```py
self.assertEqual( "-5.3", str(value_neg_str.total()) )
```

It would be a pain, though.

Reply:
I'm not sure I understand why it's not clear because it uses Decimal: that's the only type we can use to make sure that the exact value is being preserved. If you were confused about why I'm passing a string to Decimal, it's because it's the easiest way to pass in an exact value.

However, I think you're right in that the thing we're ultimately trying to test is that the right value would be printed to the document. I'll add a few of those tests where possible, but will keep most of them testing `Decimal == Decimal`.